### ISSUE TITLE
New lint: Prefer `.unwrap_or(0)` over `.unwrap_or_default()`

### ISSUE THREAD

**Issue by nik-rev** [rust-lang/rust-clippy#14779]
### What it does

This lint disallows usages of the methods using `Default` trait when there is an alternative, explicit type for a select number of types which are deemed to have an "obvious" default implementation

Methods such as:
- `Option::unwrap_or_default` -> `Option::unwrap_or`
- `Result::unwrap_or_default` -> `Result::unwrap_or`
- `Entry::or_default` -> `Entry::or_insert`

Types affected:

- numbers: `0`or `0.0`
- `&str`: `""`
- `bool`: `false`
- `char`: `'\0'`
- `Option<T>`: `None`

Thoughts? What other types or methods should be affected?

### Advantage

- Reduces cognitive complexity. You have to think about the type less
- Is more concise to type
- Everyone knows what the defaults for these types are

### Drawbacks

- Some people may prefer using `Default` trait

### Example

```rust
let a: Option<u8> = None;
let b = a.unwrap_or_default();
```

Could be written as:

```rust
let a: Option<u8> = None;
let b = a.unwrap_or(0);
```
